internal/apps/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/dd-trace-go/internal/apps
 
-go 1.22.0
+go 1.23
```

Comment:
Should this not be:
```suggestion
go 1.23.0
```

Reply:
Not sure. What I want is the latest version of go1.23. I'm not interested in pinning to `go1.23.0` specifically. That being said, I'm unsure if `go1.23` is just an alias for `go.1.23.0` here or if it actually behaves the way I want.

I don't care much either way, so unless you have a strong opinion, let's leave it as-is?

Reply:
IIRC,  `go 1.23` is "go 1.23 dev releases or newer", whereas "go 1.23.0" is "go 1.23.0 release or newer".

If you want the latest, you need to `go get go@1.23` and that'll resolve whatever the current latest is and update your `go.mod` to match that... But you probably also wanna set `GOTOOLCHAIN=local` to avoid introduction of a `toolchain` directive.

Reply:
Do you have some link to back up this interpretation of the go version directive? I can't find it in the [module spec](https://go.dev/ref/mod#go-mod-file-go).

Reply:
> IIRC, go 1.23 is "go 1.23 dev releases or newer", whereas "go 1.23.0" is "go 1.23.0 release or newer".

NVM, [this](https://go.dev/play/p/KkX12rqpSwI) should be the source of truth and confirms your claim. I'll update.

Reply:
121d0e3debdae2c20fd20dcddbc2da9ce0a7481f